src/Groups/matrices/form_group.jl

```diff
@@ -0,0 +1,629 @@
+# TODO: in this file several ways to get the preserved forms by a matrix group are implemented
+# we can choose to keep all of them or only some of them
+
+export
+    invariant_alternating_forms,
+    invariant_bilinear_forms,
+    invariant_hermitian_forms,
+    invariant_quadratic_forms,
+    invariant_sesquilinear_forms,
+    invariant_symmetric_forms,
+    isometry_group,
+    preserved_quadratic_forms,
+    preserved_sesquilinear_forms
+
+########################################################################
+#
+# From group to form #TODO: there are different approaches. Which is the best?
+#
+########################################################################
+
+# TODO 1st approach: brute force calculation
+# Algorithm furnished by Thomas Breuer, Aachen University
+# extended to quadratic by Giovanni De Franceschi, TU Kaiserslautern
+# WARNING: huge linear system !!
+
+# TODO: there are also methods to get only symmetric / alternating / hermitian forms. In these cases, the resulting linear system has lower dimension.
+
+"""
+    invariant_bilinear_forms(G::MatrixGroup)
+
+Return a generating set for the vector spaces of bilinear forms preserved by the group `G`.
+!!! warning "Note:"
+    At the moment, elements of the generating set are returned of type `mat_elem_type(G)`.
+"""
+function invariant_bilinear_forms(G::MatrixGroup{S,T}) where {S,T}
+   F = base_ring(G)
+   n = degree(G)
+   M = T[]
+   for mat in gens(G)
+      mmat = mat^-1
+      MM = zero_matrix(F,n^2,n^2)
+      for i in 1:n, j in 1:n, k in 1:n
+         MM[n*(i-1)+j,n*(k-1)+j] += mat[i,k]
+         MM[n*(i-1)+j,n*(i-1)+k] -= mmat[j,k]
+      end
+      push!(M,MM)
+   end
+
+   r,K = nullspace(block_matrix(length(M),1,M))
+   
+   return [matrix(F,n,n,[K[i,j] for i in 1:n^2]) for j in 1:r]
+end
+
+"""
+    invariant_sesquilinear_forms(G::MatrixGroup)
+
+Return a generating set for the vector spaces of sesquilinear non-bilinear forms preserved by the group `G`. It works only if `base_ring(G)` has even degree.
```

Comment:
... what's the degree of an arbitrary ring? I think you mean to say that `base_ring(G)` must be a finite field with an involution, right?

Reply:
This kind of forms can be defined over any field with an involutory automorphism (e.g. the complex field). But these methods work only on finite fields. I will put a restriction on the type.

Reply:
I know -- my questions are kinda rhetorical. Let me rephrase then: "This text is confusing because it talks about the degree of a ring, which is AFAIK not a general concept. I think what you really want to say here is that this method expects the base ring to be a finite field which has even degree over its prime field. Then you should say that, and also restrict the signature accordingly."

Well, or else, make the function generic, as I then suggested a few lines later...